src/rpc/handlers/AccountObjects.cpp

```diff
@@ -56,49 +56,50 @@ doAccountNFTs(Context const& context)
     if (auto const status = getLimit(request, limit); status)
         return status;
 
-    ripple::uint256 marker;
-    if (auto const status = getHexMarker(request, marker); status)
+    ripple::uint256 pageMarker;
+    ripple::uint256 tokenMarker;
+    if (auto const status =
+            getAccountNFTMarker(request, pageMarker, tokenMarker);
+        status)
         return status;
+    ripple::uint256 const maskedMarker = tokenMarker & ripple::nft::pageMask;
 
-    auto const first =
-        ripple::keylet::nftpage(ripple::keylet::nftpage_min(accountID), marker);
-    auto const last = ripple::keylet::nftpage_max(accountID);
-
-    auto const key =
-        context.backend
-            ->fetchSuccessorKey(first.key, lgrInfo.seq, context.yield)
-            .value_or(last.key);
-    auto const blob = context.backend->fetchLedgerObject(
-        ripple::Keylet(ripple::ltNFTOKEN_PAGE, key).key,
-        lgrInfo.seq,
-        context.yield);
-
-    std::optional<ripple::SLE const> cp{
-        ripple::SLE{ripple::SerialIter{blob->data(), blob->size()}, key}};
+    response[JS(account)] = ripple::toBase58(accountID);
 
     std::uint32_t cnt = 0;
     response[JS(account_nfts)] = boost::json::value(boost::json::array_kind);
     auto& nfts = response.at(JS(account_nfts)).as_array();
 
-    bool pastMarker = marker.isZero();
-    ripple::uint256 const maskedMarker = marker & ripple::nft::pageMask;
+    // start iteration at the last page, which is guaranteed to exist if the
+    // account owns any NFTs
+    auto const last = ripple::keylet::nftpage_max(accountID);
+    bool hasMarker = !pageMarker.isZero();
+    auto const key = hasMarker ? pageMarker : last.key;
+
+    auto const blob =
+        context.backend->fetchLedgerObject(key, lgrInfo.seq, context.yield);
+    if (!blob)
+        return response;
+    std::optional<ripple::SLE const> cp{
+        ripple::SLE{ripple::SerialIter{blob->data(), blob->size()}, key}};
 
     // Continue iteration from the current page:
-    while (true)
+    while (cp)
     {
         auto arr = cp->getFieldArray(ripple::sfNFTokens);
 
         for (auto const& o : arr)
         {
             ripple::uint256 const nftokenID = o[ripple::sfNFTokenID];
+
             ripple::uint256 const maskedNftokenID =
                 nftokenID & ripple::nft::pageMask;
 
-            if (!pastMarker && maskedNftokenID < maskedMarker)
+            if (hasMarker && maskedNftokenID < maskedMarker)
                 continue;
 
-            if (!pastMarker && maskedNftokenID == maskedMarker &&
-                nftokenID <= marker)
+            if (hasMarker && maskedNftokenID == maskedMarker &&
+                nftokenID <= tokenMarker)
```

Comment:
Yup, these look right.

I'll mention that the reason I used `bool pastMarker` in my own implementation was to short circuit these `uint256` comparisons.  We know that the entries in the page are sorted (albeit oddly), so once we're past the marker we no longer need to do the comparisons.  In the code as currently written the `maskedNftokenID` comparisons will happen every time, even once you have passed the marker.  So the code as it stands is correct, but it's a little less efficient than it might be.